packages/flutter_tools/test/channel_test.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/command_runner.dart';
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/channel.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:test/test.dart';
+
+import 'src/common.dart';
+import 'src/context.dart';
+
+void main() {
+  group('channel', () {
+    Logger oldLogger;
+    BufferLogger logger;
+
+    setUp(() {
+      Cache.disableLocking();
+      logger = new BufferLogger();
+      Cache.flutterRoot = '../..';
+      context[DeviceManager] = new MockDeviceManager();
+      oldLogger = context[Logger];
+      context[Logger] = logger;
+    });
+
+    tearDown(() {
+      context[Logger] = oldLogger;
+    });
+
+    test('list', () async {
```

Comment:
I think you can get rid of the setup and teardown methods if you change `test(` to `testUsingContext(`. See create_test.dart or driver_test.dart. Within the testUsingContext closure, you can assume that the `logger` is an instance of `BufferLogger`.